src/observability/kubearmor.go

```diff
@@ -68,11 +68,15 @@ func clearKubeArmorLogMap() {
 }
 
 func ProcessSystemLogs() {
+
+	SystemLogsMutex.Lock()
 	if len(SystemLogs) <= 0 {
+		// we are using Unlock instead of defer Unlock to immediately release the variable in order to
+		// prevent the log channel from filling up
+		SystemLogsMutex.Unlock()
 		return
 	}
```

Comment:
remove lock/unlock sphaghetti.

```suggestion

	SystemLogsMutex.Lock()
	locSysLogs := SystemLogs
	SystemLogs = []*pb.Alert{} //reset
	SystemLogsMutex.Unlock()

	if len(locSysLogs) <= 0 {
		return
	}
```